storage/Devices/MsdosImpl.cc

```diff
@@ -63,6 +63,29 @@ namespace storage
 
 
     void
+    Msdos::Impl::delete_partition(Partition* partition)
+    {
+	PartitionType old_type = partition->get_type();
+	unsigned int old_number = partition->get_number();
+
+	PartitionTable::Impl::delete_partition(partition);
+
+	// After deleting a logical partition the numbers of partitions with
+	// higher numbers are shifted.
+	if (old_type == PartitionType::LOGICAL)
+	{
+	    vector<Partition*> partitions = get_partitions();
+            for (Partition* tmp : partitions)
+	    {
+		unsigned int number = tmp->get_number();
+		if (number > old_number)
```

Comment:
There are really two checks here (or rather, there should be two): One for the starting point from where to renumber, and one for the new expected number. Since it's all rolled into one due to the iteration, it might make sense to calculate the new expected number for each one, and if the actual number is different from that one, change it (which might involve more complex things like renaming them in files like /etc/crypttab and /etc/fstab).

Admittedly, if everything has been correct before, the outcome should be the same. But I don't like to rely on things being correct before.